### ISSUE TITLE
Prometheus error: Address already in use

### ISSUE THREAD

**Issue by CharlieC3** [stacks-network/stacks-core#4214]
**Describe the bug**
When running [this commit](https://github.com/stacks-network/stacks-core/commit/c8db64c59ad9d95b53dbc37d529da4b43043fac3) from the `clarity-wasm-next` branch, the stacks node emits an error when booting using a Bitcoin node in regtest mode.

```
2023-12-26T10:20:50-05:00	Process abort due to thread panic: panicked at testnet/stacks-node/src/monitoring/prometheus.rs:13:14:
Prometheus monitoring: unable to bind address: Os { code: 98, kind: AddrInUse, message: "Address already in use" }
2023-12-26T10:20:50-05:00	Start serving prometheus metrics
```

The node doesn't crash from this error, and the Prometheus metrics still seem to be getting served on the configured port. So perhaps the Prometheus server is trying to start twice?

**Steps To Reproduce**
Run the `clarity-wasm-next` branch in a regtest environment. You can do so by cloning the [regtest project](https://github.com/hirosystems/stacks-regtest-env/tree/epoch3-interim-blocks) (epoch3-interim-blocks branch) and modifying [this variable](https://github.com/hirosystems/stacks-regtest-env/blob/epoch3-interim-blocks/docker-compose.yml#L4C26-L4C26) to the `clarity-wasm-next` branch.

**Comment by CharlieC3** [comment]
FYI @obycode

**Comment by CharlieC3** [comment]
I see this error occurring on Nakamoto-1 as well (cc @kantai). Priority is low, since this just seems to cause a single crash, then it continues normally on the next reboot.

**Comment by kantai** [comment]
Yep -- this is a bug in the nakamoto run-loop: it tries to start the prometheus server, but if the neon loop started the prometheus server already, it will panic.

**Comment by kantai** [comment]
This should be resolved by the merge of #4226.
